### PR TITLE
Updates to the Dev VM environment/tooling

### PR DESCRIPTION
**Pull request by marcparadise**
This makes many changes to improve the Dev VM environment:

* remove 'sync' tool, because native/direct file share with host
  is now performant enough on both Mac and Linux
* removed unsupported scenarios and code from Vagrantfile
* clarify Vagrantfile contents via renames and comments
* fix project loading for erlang and ruby projects
* fix project loading for cookbooks and ctl commands
* fix erlang project depenency handling and hotloading
* update README to reflect latest changes
* install some additional tooling (tree, jq, ripgrep)
* update distro to ubuntu 20.04
* remove 'quickstart' behaviors which weren't used and didn't always
  work very well.
* set up a default org and admin user; ensure `knife` works out of the
  box for this configuration for root.
* with the sync tool gone, remove dvmtools.rb and move the remaining
  code into the Vagrantfile

Signed-off-by: Marc A. Paradise <marc.paradise@gmail.com>

### CODE REVIEW

File: dev/README.md [hunk 1]
```diff
@@ -68,11 +68,11 @@ Now on your workstation, create `.chef/knife.rb` in the root of your
 chef-server checkout, with the following:
 
     current_dir = File.dirname(__FILE__)
-	log_level                :info
-	log_location             STDOUT
-	node_name                "admin"
-	client_key               "#{current_dir}/admin.pem"
-	chef_server_url          "https://api.chef-server.dev/organizations/test"
+	  log_level                :info
```
Comment (tas50) [inline dev/README.md:71]:
```suggestion
```

File: dev/README.md [hunk 2]
```diff
@@ -68,11 +68,11 @@ Now on your workstation, create `.chef/knife.rb` in the root of your
...
+	  log_location             STDOUT
```
Comment (tas50) [inline dev/README.md:72]:
```suggestion
```

**Review by PrajaktaPurohit — APPROVED** [review]

File: dev/README.md [hunk 3]
```diff
@@ -206,55 +213,55 @@ And finally to disable coverage and re-enable automatic code sync:
...
-`irb(main):001:0> `which vagrant`
-=> "/opt/chefdk/embedded/bin/vagrant\n"
-```
+## Erlang Project Dependencies
 
-You can tell `./sync` what Vagrant to use by setting `$VAGRANT_PATH` like so:
+If you need to work on an erlang project dependency (for example, you want to modify chef_authn)
+this can also be loaded into the VM with hot code loading support.  After loading the service
+in the VM (eg. `dvm load oc_erchef`) , switch to the host and clone the dependency into `external-deps`
+directory. For example, to load `chef_authn` into a loaded and running `oc_erchef`:
```
Comment (PrajaktaPurohit) [inline dev/README.md:239]: Thank you for writing this up!

**Review by PrajaktaPurohit — APPROVED** [review]

File: dev/README.md [hunk 4]
```diff
@@ -83,19 +83,16 @@ and/or chef-server-ctl commands.
 
 * Changes to erchef erlang files will be picked up and recompiled
   automatically shortly after you save them on the host.
-* To test cookbook changes, load them with `dvm load omnibus private-chef-cookbooks`.
-  Then run `chef-server-ctl reconfigure` in the VM to pick up the changes.
-* upgrades and chef-server-ctl command changes/additions will be
-  available very quickly after you save them on the host (< 5 seconds)
+* To test cookbook changes, load them with `dvm load omnibus server-ctl-cookbooks`
+  run `chef-server-ctl reconfigure` in the VM as needed to pick up the changes.
```
Comment (jashaik) [inline dev/README.md:87]: It is looking like `dvm load omnibus server-ctl-cookbooks` is also triggering the `chef-server-ctl reconfigure`.

**Comment by jashaik** [comment]
### Testing Report:

- use virtualbox native sharing to access host files.
- remove the ‘sync’ tool  -> sync of files is working without sync lib
- project loading from host is fixed for:
- oc-id -> working
- bifrost -> working
- erchef -> working
- bookshelf -> working
- server cookbooks -> working
- server ctl commands  -> working
- hot code updates from host works for erlang projects -> working
- automatically set up an org with an admin user -> Working
- fix path so that chef/knife work by default  -> working 
- install additional default packages: tree, jq, ripgrep -> installed
- remove broken ‘quickstart’ support -> removed this with load & start
- update readme to reflect any usage changes -> Need to review

@PrajaktaPurohit @vinay-satish @marcparadise

**Review by jashaik — APPROVED** [review]

**Pull request merged**